behavior_namespace: "/clickable_world"
behavior_names: ["table_approach", "move_floor", "grasp_object", "place_object"]
#behavior_names: ["grasp_object"]
#behavior_names: ["table_approach", "move_floor"]
#behavior_names: ["table_approach", "move_floor", "grasp_object"]
#behavior_names: ["test_button"]
#behavior_names: ["move_floor"]
init_conditions: {"at_table" : False, "obj_in_hand" : False}
#init_conditions: {"at_table" : True, "obj_in_hand" : False}
#init_conditions: {"at_table" : True, "obj_in_hand" : True}

test_button:
    preconditions: {}
    postconditions: {}
    perception_srv: "/test_button_perception"
    action_srv: "/test_button_action"

move_floor:
    preconditions: {"at_table" : False}
    postconditions: {}
    perception_srv: "/detect_empty_floor"
    action_srv: "/move_empty_floor"

table_approach:
    preconditions: {"at_table" : False}
    postconditions: {"at_table" : True}
    perception_srv: "/table_perception"
    action_srv: "/table_approach_action"

table_backoff:
    preconditions: {"at_table" : True}
    postconditions: {"at_table" : False}
    perception_srv: "/backoff_button"
    action_srv: "/backoff_table"

grasp_object:
    preconditions: {"at_table" : True, "obj_in_hand" : False}
    postconditions: {"obj_in_hand" : True}
    perception_srv: "/detect_objects"
    action_srv: "/grasp_object"

place_object:
    preconditions: {"at_table" : True, "obj_in_hand" : True}
    postconditions: {"obj_in_hand" : False}
    perception_srv: "/place_table_perceive"
    action_srv: "/place_object"

handoff_object:
    preconditions: {"at_table" : False, "obj_in_hand" : True}
    postconditions: {"obj_in_hand" : False}
    perception_srv: "/detect_faces"
    action_srv: "/handoff_obj_to_person"
    
